frontend/src/components/PractitionerFeedListItemEdit.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useContext, useState } from "react";
+import { getLabelNameFromPractitionerResult } from '../utils/helpers';
+import { HapiPractitionerEntryItemParams } from './PractitionerFeedListItem';
+import { HapiPractitionersResponse, HapiTelecomType } from '../types/Hapi.types';
+import axios, { AxiosResponse } from 'axios';
+import { HapiPractitionersFetchRequestUrl } from '../constants';
+import { AppContext } from '../AppContext';
+import { Button, Input, ListItemText } from '@mui/material';
+
+interface PractitionerFeedListItemEdit extends HapiPractitionerEntryItemParams { setShowEdit: any }
+
+const PractitionerFeedListItemEdit: React.FC<PractitionerFeedListItemEdit> = ({ practitioner, setShowEdit }: PractitionerFeedListItemEdit): JSX.Element => {
+  const [input, setInput] = useState({
+    firstName: practitioner.firstName,
+    lastName: practitioner.lastName,
+    phone: practitioner.phone,
+    fax: practitioner.fax,
+    email: practitioner.email,
+    address: practitioner.address,
+  })
+
+  const appContext = useContext(AppContext);
+
+  const payload = {
+    resourceType: "Practitioner",
+    id: practitioner.id,
+    identifier: [
+      {
+        system: "http://fhir.de/NamingSystem/kbv/lanr",
+        value: practitioner.id
+      }
+    ],
+    name: [
+      {
+        use: "usual",
+        family: input.lastName,
+        given: input.firstName,
+        prefix: ["Dr."]
+      },
+      {
+        _family: {
+          extension: [
+            {
+              url: "http://hl7.org/fhir/StructureDefinition/humanname-own-name",
+              valueString: "Test 3"
+            }
+          ]
+        }
+      }
+    ],
+    telecom: [
+      {
+        system: HapiTelecomType.PHONE,
+        value: input.phone
+      },
+      {
+        system: HapiTelecomType.FAX,
+        value: input.fax
+      },
+      {
+        system: HapiTelecomType.EMAIL,
+        value: input.email
+      }
+    ],
+    address: [
+      {
+        text: input.address
+      }
+    ]
+  }
+
+  const submitInfo = () => {
+    axios.put(`${HapiPractitionersFetchRequestUrl}/${practitioner.id}`, payload,{ headers: { 'Content-Type': 'application/fhir+json' }})
+      .then((res: AxiosResponse<HapiPractitionersResponse>) => {
+        appContext?.dispatch({
+          type: 'update-practitioners',
+          payload: res.data
+        });
+        setShowEdit();
+      })
+      .catch((err) => {
+        alert(err.message);
+      });
+  }
+
+  return (
+      <form>
```

Comment:
I appreciate that you are going for a lightweight implementation here. I'd imagine that, were you to have more time, you would use or develop a form management library with validation, etc. 

It's something I'd expect here, though (validation particularly).

Reply:
yes it would've better to use a form management library for sure. should've spent a little more time on this area. validation is definitely a boilerplate thing.